src/utilities/routeRegex.spec.ts

```diff
@@ -28,7 +28,7 @@ describe('generateRoutePathRegexPattern', () => {
 
     const result = generateRoutePathRegexPattern(routerRoutes)
 
-    const catchAll = '([^/]+)'
+    const catchAll = '[^/]+'
```

Comment:
previously routeRegex was using capture groups unnecessarily. Needed to be non-capturing to reuse in paramFinder